R/scale_marker.R

```diff
@@ -0,0 +1,44 @@
+
+
+scale_marker_discrete <- function(glyphs = NULL, colors = NULL, limits = NULL, ...) {
+
+  markers <- data.frame(glyphs = glyphs, colors = colors, labels = limits) |>
+    distinct()
+
+  palette <- pal_markers(
+    glyphs = markers$glyphs,
+    colors = markers$colors,
+    n_values = nrow(markers)
+  )
+
+  discrete_scale("marker", rlang::missing_arg(), palette = palette,
+                 limits = markers$labels, ..., na.translate = FALSE)
```

Comment:
`na.translate` is needed here because the `<vctr_rcrd>` doesn't work well with an `ifelse()` statement somewhere in the scale mapping.